File: example/cpp/picsocket_sender.cpp
/*
 * Copyright (C) 2024 Coder.AN
 * Email: an.hongjun@foxmail.com
 * Page: www.anhongjun.top
 */
 #include <cstdlib>
 #include <cstdio>
 #include "pic_socket.h"

 int main(int argc, char* argv[])
 {
    if (argc != 3){
        printf("Usage: %s <target-ip> <target-port>\n", argv[0]);
        return -1;
    }
    
    cv::VideoCapture capture(0);
    UDPImgSender img_sender(argv[1], atoi(argv[2]));

    while (true)
	{
		cv::Mat frame;
		capture >> frame;
        img_sender.send(frame);
		cv::imshow("sender", frame);
		cv::waitKey(30);
	}
	return 0;
 }

File: example/python/picsocket_sender.py
# Copyright (C) 2024 Coder.AN
# Email: an.hongjun@foxmail.com
# Page: www.anhongjun.top

import sys
import cv2
import picsocket


if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Usage: {} <target-ip> <target-port>".format(sys.argv[0]))
    img_sender = picsocket.ImgSender(sys.argv[1], int(sys.argv[2]))

    cap = cv2.VideoCapture(0)
    while True:
        ret, img = cap.read()
        if img is None:
            print("Read images error!")
            break
        img_sender.send(img)
